api/v1alpha1/linodecluster_webhook_test.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 Akamai Technologies, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/linode/cluster-api-provider-linode/mock"
+
+	. "github.com/linode/cluster-api-provider-linode/mock/mocktest"
+)
+
+func TestValidateLinodeCluster(t *testing.T) {
+	t.Parallel()
+
+	var (
+		cluster = LinodeCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example",
+				Namespace: "example",
+			},
+			Spec: LinodeClusterSpec{
+				Region: "example",
+			},
+		}
+	)
+
+	NewSuite(t, mock.MockLinodeClient{}).Run(
+		OneOf(
+			Path(
+				Call("valid", func(ctx context.Context, mck Mock) {
+					mck.LinodeClient.EXPECT().GetRegion(gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
+				}),
+				Result("success", func(ctx context.Context, mck Mock) {
+					assert.NoError(t, cluster.validateLinodeCluster(ctx, mck.LinodeClient))
+				}),
+			),
+		),
+		OneOf(
+			Path(Call("invalid region", func(ctx context.Context, mck Mock) {
+				mck.LinodeClient.EXPECT().GetRegion(gomock.Any(), gomock.Any()).Return(nil, errors.New("invalid region")).AnyTimes()
+			})),
+		),
+		Result("error", func(ctx context.Context, mck Mock) {
+			assert.Error(t, cluster.validateLinodeCluster(ctx, mck.LinodeClient))
+		}),
```

Comment:
We should also validate the error messages like we do [here](https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodecluster_controller_test.go#L122)

Reply:
If possible, I would prefer to keep these tests generic against the `Linode*`resource interface and decoupled from the the implementation details.

In this case, these error messages are more of a convenience for the user to help them triage the exact validation error rather than semantically meaningful to the controller itself.

Let me know if you'd prefer otherwise though.

Reply:
For this specific implementation, we should be ok with not looking for the error string because the method has just one error check. 

I like doing those checks in cases where we have the same method checking for errors at multiple stages like in our controllers. So we want to make sure that we are checking for the specific error to have occurred rather than just some/any error to have occurred